### DISCUSSION TITLE
Any fast scripts for a blurred background when running i3lock?

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#5871] [Q&A]
I currently have this script:
```
#!/bin/bash

scrot /tmp/screenshot.png
convert /tmp/screenshot.png -blur 0x9 /tmp/screenshotblur.png

i3lock -i /tmp/screenshotblur.png
```

but it takes around 5 seconds to actually work. Does anyone have any better methods?

**Comment by orestisfl** [comment]
See i3lock-color: https://github.com/Raymo111/i3lock-color

**Reply by orestisfl** [reply]
Alternatively, to immediately lock the screen, you could adjust your script so that:
1. Immediately spawn i3lock without an image in the background
2. Do the scrot & convert steps
3. Kill the first i3lock and start a new one with the `/tmp/screenshotblur.png` image

**Comment by neve-dimka** [comment]
```
convert /home/user/pictures/wallpaper.png -blur 0x9 /home/user/pictures/blurred_wallpaper.png
#!/bin/bash
i3lock -i /home/user/pictures/blurred_wallpaper.png
```
if you have script that constantly change your wallpaper, make it constantly update blurred_wallpaper.png then.
